independent-projects/bootstrap/maven-plugin/src/main/java/io/quarkus/maven/ExtensionDescriptorMojo.java

```diff
@@ -274,16 +283,50 @@ private void transformLegacyToNew(final Path output, ObjectNode extObject, Objec
             metadata.set("short-name", extObject.get("shortName"));
             extObject.remove("shortName");
         }
-
+        // Add quarkus core version used to build this extension
+        String quarkusCoreVersion = resolveQuarkusCoreVersion();
+        if (quarkusCoreVersion != null) {
+            metadata.put("quarkus-core", quarkusCoreVersion);
+        }
         extObject.set("metadata", metadata);
+    }
 
-        //   updateSourceFiles(output, extObject, mapper);
+    private String resolveQuarkusCoreVersion() throws MojoExecutionException {
+        MavenArtifactResolver mvn = getMavenResolver();
+        DefaultArtifact artifact = new DefaultArtifact(project.getGroupId(),
+                project.getArtifactId(),
+                project.getPackaging(),
```

Comment:
'packaging' is not the same as the type. `projectArtifact.getArtifactHandler().getExtension()` will be more appropriate.